server/service.go

```diff
@@ -343,8 +360,88 @@ func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request)
 		"relays":      strings.Join(types.RelayEntriesToStrings(result.relays), ", "),
 	}).Info("best bid")
 
-	// Return the bid
-	m.respondOK(w, &result.response)
+	// Define map of supported handlers
+	supportedMediaTypeHandlers := map[string]func(){
+		MediaTypeJSON: func() {
```

Comment:
i find this nested pattern confusing, and it makes the whole function complex.

could you break the two out as individual functions? maybe the map also isn't needed, just in the end you request the right function? would simplify things